src/ipc/ipc4/helper.c

```diff
@@ -789,3 +789,19 @@ int ipc4_add_comp_dev(struct comp_dev *dev)
 
 	return IPC4_SUCCESS;
 };
+
+int ipc4_find_dma_config(struct ipc_config_dai *dai, uint8_t *data_buffer, uint32_t size)
+{
+#if defined(CONFIG_ACE_VERSION_2_0)
+	struct ipc_dma_config *dma_config =
+	    (struct ipc_dma_config *)GET_IPC_DMA_CONFIG(data_buffer, size);
+	uint32_t *dma_config_id = GET_IPC_DMA_CONFIG_ID(data_buffer, size);
+
+	if (*dma_config_id != GTW_DMA_CONFIG_ID)
+		return IPC4_INVALID_REQUEST;
+
+	dai->host_dma_config = dma_config;
+#endif
+	return IPC4_SUCCESS;
+}
```

Comment:
Are we going to re-use ipc4_find_dma_config for MTL or TGL?
If there are no plans for it and this is purely LNL function I would wrap whole function definition with #ifdef (so we do not produce more dummy functions) and also calls to it.

Reply:
it's my understanding that the DMA_CONFIG_TLV is only relevant for ACE2+. It's useless for MTL and TGL.